schema/m3-region.schema.json

```diff
@@ -346,9 +397,50 @@
               "required": [],
               "additionalProperties": false,
               "properties": {}
-            }    
+            },
+            "leaveWithGModeSetup": {
+              "$id": "#/definitions/strat/properties/exitCondition/properties/leaveWithGModeSetup",
+              "type": "object",
+              "title": "Leave With G-Mode Setup",
+              "description": "Represents that Samus can leave through this door while taking damage through the transition, in a pose that would allow using X-Ray on the first frame after the transition. This is for setting up to enter R-mode or direct G-mode in the next room.",
+              "required": [],
+              "additionalProperties": false,
+              "properties": {
+                "knockback": {
+                  "$id": "#/definitions/strat/properties/exitCondition/properties/leaveWithGModeSetup/properties/knockback",
+                  "type": "boolean",
+                  "default": true,
+                  "title": "Leaves With Knockback",
+                  "description": "If true, then Samus gets knockback frames through the transition."
+                }
+              }
+            },
+            "leaveWithGMode": {
+              "$id": "#/definitions/strat/properties/exitCondition/properties/leaveWithGMode",
+              "type": "object",
+              "title": "Leave With G-Mode",
+              "description": "Represents that Samus can leave through this door while in G-mode, resulting in indirect G-mode in the next room.",
+              "required": [],
+              "additionalProperties": false,
+              "properties": {
+                "morphed": {
+                  "$id": "#/definitions/strat/properties/exitCondition/properties/leaveWithGMode/properties/morphed",
+                  "type": "boolean",
+                  "title": "Morphed",
+                  "description": "If true, then this strat results in leaving the room in a morphed state, either by maintaining artificial morph or by having the Morph item."
+                }
+              }
+            }
           }
         },
+        "gModeRegainMobility": {
+          "$id": "#/definitions/strat/properties/gModeRegainMobility",
+          "type": "object",
+          "description": "Indicates that this strat allows Samus to regain mobility (by taking knockback from an enemy or projectile) after entering with G-mode immobile.",
```

Comment:
So the thing is that these strats describe how to get out of the immobile state while staying in G-mode, i.e. how to regain mobility by taking an enemy hit. They don't describe that you can enter G-Mode Immobile at that location, because that is something that could be done at any door (hence doesn't need to be explicitly described in the data); it's just that entering with G-Mode Immobile isn't useful without a way to get out of it while retaining G-mode. So to me "RegainMobility" still seems more accurate for what these strats describe. But I don't care too much about the name either way; what do you think @kjbranch ?

(Also updated the description here to remove the reference to projectile, and clarify about staying in G-mode.)